src/pages/en/core-concepts/astro-components.md

```diff
@@ -125,22 +125,59 @@ They can be used to style your components, and all style rules are automatically
 
 ### Client-Side Scripts
 
-To send JavaScript to the browser without [using a framework component](/en/core-concepts/component-hydration) (React, Svelte, Vue, Preact, Solid...) you can use a `<script>` tag in your Astro component body.
+To send JavaScript to the browser without [using a framework component](/en/core-concepts/framework-components) (React, Svelte, Vue, Preact, SolidJS, AlpineJS, Lit...) you can use a `<script>` tag in your Astro component template and send JavaScript to the browser that executes in the global scope.
 
 ```astro
 ---
-// Your component script here!
+// Example: Using Astro with script tags
 ---
+<h1>Not clicked</h1>
+<button>Click to change heading</button>
+
 <script>
-  document.querySelector('h1').style.color = 'red';
+document.querySelector("button").addEventListener("click",() => {
+    document.querySelector("h1").innerText = "clicked"
+})
 </script>
+```
+ > ⚠️ Starting in `--experimental-static-build` (v0.23.x), you must opt in to `<script>` element processing via the `hoist` attribute:
```

Comment:
I think it's safe to remove lines 130-143. The "Starting in `--experimental-static-build`" example should work across with/without flag, and since it's the direction that we want to be going I think you can just document this for everyone.

Reply:
Done!